core/src/main/java/com/novoda/noplayer/mediaplayer/AndroidMediaPlayerFacade.java

```diff
@@ -61,8 +66,7 @@ void setSurfaceHolderRequester(SurfaceHolderRequester surfaceHolderRequester) {
 
     void prepareVideo(final Uri videoUri) {
         if (surfaceHolderRequester == null) {
-            logPlayerNotAttachedWarning("prepareVideo()");
-            return;
+            throw new IllegalStateException("Must set a SurfaceHolderRequester before preparing video");
```

Comment:
If it's necessary to set all these things before a AndroidMediaPlayerFacade instance can be used, can we pass them as newInstance parameters?